types/got/index.d.ts

```diff
@@ -113,16 +115,20 @@ declare namespace got {
     }
 
     interface GotOptions<E extends string | null> extends InternalRequestOptions {
+        baseUrl?: string;
+        headers?: { [key: string]: string };
+        cookieJar?: CookieJar;
         encoding?: E;
         query?: string | object;
         timeout?: number | TimeoutOptions;
-        retries?: number | RetryFunction;
+        retry?: number | RetryOptions;
         followRedirect?: boolean;
         decompress?: boolean;
         useElectronNet?: boolean;
-        cache?: Cache;
-        agent?: http.Agent | boolean | AgentOptions;
         throwHttpErrors?: boolean;
+        agent?: http.Agent | boolean | AgentOptions;
+        hooks?: { [key: string]: Array<() => any> };
```

Comment:
As I understand from [documentation](https://github.com/sindresorhus/got#hooks) hooks are fixed. Therefore, more narrow type can be used here:

```ts

type Hook = () => any;

hooks?: Record<'beforeRequest', Hook>;
```

Also hook function takes options as argument.